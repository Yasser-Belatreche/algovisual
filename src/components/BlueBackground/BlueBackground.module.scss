.container {
  @apply relative;

  &::after {
    @apply w-screen bg-blue h-full top-2/4 left-2/4  absolute;
    content: "";
    transform: translate(-50%, -50%);
    z-index: -1;
  }
}
