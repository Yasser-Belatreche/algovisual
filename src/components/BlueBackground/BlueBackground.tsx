import React from "react";

// styles
import styles from "./BlueBackground.module.scss";

interface Props extends React.HTMLAttributes<HTMLDivElement> {}

const BlueBackground: React.FC<Props> = ({
  children,
  className: classFromProps,
  ...props
}) => {
  return (
    <div
      className={`${styles.container} ${classFromProps}`}
      {...props}
      role="container"
    >
      {children}
    </div>
  );
};

export { BlueBackground };
