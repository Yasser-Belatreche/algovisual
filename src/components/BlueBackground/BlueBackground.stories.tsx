import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { BlueBackground } from "./BlueBackground";

export default {
  title: "Global/BlueBackground",
  component: BlueBackground,
  argTypes: {},
} as ComponentMeta<typeof BlueBackground>;

const Template: ComponentStory<typeof BlueBackground> = (args) => (
  <BlueBackground {...args}>
    <div
      style={{
        padding: "100px",
        color: "white",
        fontSize: 30,
        textAlign: "center",
      }}
    >
      Some content
    </div>
  </BlueBackground>
);

export const BlueBackgroundDefault = Template.bind({});
BlueBackgroundDefault.args = {};
