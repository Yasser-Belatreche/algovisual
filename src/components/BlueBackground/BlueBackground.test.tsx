import { render } from "@testing-library/react";
import { BlueBackground } from "./BlueBackground";

describe("BlueBackground component", () => {
  it("should render a blue background that take the full screen with around his children", () => {
    const { container } = render(
      <BlueBackground>Some children</BlueBackground>
    );
    expect(container).toMatchSnapshot();
  });

  it("the container div should have the props passed to the component", () => {
    const props = {
      className: "someClass",
      onClick: () => {},
    };
    const instance = render(<BlueBackground {...props}></BlueBackground>);

    const container = instance.getByRole("container");

    expect(container.classList.length).toEqual(2);
    expect(container.classList).toContain(props.className);
    expect(container.onclick).toBeDefined();

    expect(instance.container).toMatchSnapshot();
  });
});
