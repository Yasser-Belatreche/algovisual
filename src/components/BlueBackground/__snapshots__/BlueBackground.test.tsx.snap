// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlueBackground component should render a blue background that take the full screen with around his children 1`] = `
<div>
  <div
    class="container undefined"
    role="container"
  >
    Some children
  </div>
</div>
`;

exports[`BlueBackground component the container div should have the props passed to the component 1`] = `
<div>
  <div
    class="container someClass"
    role="container"
  />
</div>
`;
